src/qtfirebase.cpp

```diff
@@ -60,6 +63,7 @@ void QtFirebase::waitForFutureCompletion(firebase::FutureBase future)
             count = 0;
             break;
         }
+        QThread::msleep(10);
```

Comment:
I'm a bit unsure why you've put this here? :)

Reply:
It lowers the CPU usage and gives a better way to cancel the loop (more independent of CPU speed). Also (I forgot to change this): the 100000 and 20000 numbers must be reduced to something like 100 (1 second) and 200 (2 seconds).